themes/default/content/blog/next-level-iac-package-ecosystems/index.md

```diff
@@ -0,0 +1,137 @@
+
+The flexibility that Pulumi gives you, and the access to these powerful libraries of code, means your productivity can skyrocket when you get to write in your favorite language, using your preferred tools and the package ecosystem you already know, love, and rely on every day.
+
+Let’s have a look at a concrete example where this package ecosystem can get you out of a tough spot.
+
+## Use Case: Determining S3 MIME types on-the-fly
+
+A popular example of cloud infrastructure automation is creating a S3 bucket and pushing some files to it. This is a common task that many of us have had to do, and most infrastructure-automation tools have a way to accomplish this. 
+
+Here’s our public answer for [how to do that in Pulumi using TypeScript][s3-bucket-example]: 
+
+```typescript
+import * as pulumi from '@pulumi/pulumi';
+import * as aws from '@pulumi/aws';
+import * as fs from 'fs';
+
+// Create an S3 bucket where we will upload the file.
+const bucket = new aws.s3.Bucket("my-bucket", {
+    // You can apply additional bucket settings here.
+});
+
+const filePath = "path/to/your/local/file.ext"; // Replace with the path to the file you want to upload
+
+// Read the contents of the file to be uploaded to S3.
+const fileContents = fs.readFileSync(filePath, {encoding: 'utf-8'});
+
+// Upload the file to the S3 bucket.
+const bucketObject = new aws.s3.BucketObject("my-bucket-object", {
+    bucket: bucket, // Reference to the bucket created above.
+    key: "file.txt", // This will be the file's name in the bucket.
+    content: fileContents, // Contents of the file.
+    contentType: "text/plain", // MIME-type of the file. Adjust if you upload a different type of file.
+});
+
+// Export the name of the bucket
+export const bucketName = bucket.id;
+
+// Export the URL of the file in the S3 bucket
+export const objectUrl = pulumi.interpolate`s3://${bucket.id}/${bucketObject.key}`;
+``` 
+
+However, something that stands out as a weak link here is the line:
+
+```typescript
+contentType: "text/plain", // MIME-type of the file. Adjust if you upload a different type of file.
+```
+
+Yeah, hard-coding the MIME type for every file I need to upload is a not going to scale. In fact, I’m probably going to get a random directory full of files I need to iterate over, each with different file types, and it will change every time it runs, and we won’t know what type of file it is ahead of time.
+
+Packages to the rescue!! Turns out that determining the MIME type of a file in TypeScript only involves importing the community-created [`mime-types`][npm-mimetypes] library and a single function call to `mime.lookup(...)`:
+
+```typescript
+import * as mime from 'mime-types'
+
+const mimeType = mime.lookup(‘path/to/your/local/file.ext’);
+
+```
+
+So, to work that into our Pulumi program from above, we only need to make two small edits:
+
+```typescript {hl_lines=[4,21]}
```

Comment:
I always forget we can do this! The highlighting does seem to be making the code a bit hard to read, though, and for me at least, interrupting with the rest of the syntax highlighting that follows it. Not something that need block this PR necessarily, but would be great to get something filed to handle line highlighting more beatifully!

Reply:
I agree the CSS should be tweaked cause that yellow is a little intense.